src/main/java/com/celements/course/service/CourseService.java

```diff
@@ -367,15 +368,17 @@ public RegistrationState getConfirmState(DocumentReference regDocRef) {
       Set<ParticipantStatus> status = fetcher.iter().transformAndConcat(new FieldGetterFunction<>(
           xObjFieldAccessor, CourseParticipantClass.FIELD_STATUS)).toSet();
       if (status.contains(ParticipantStatus.confirmed)) {
-        if (!status.contains(ParticipantStatus.unconfirmed)) {
-          regState = RegistrationState.CONFIRMED;
-        } else {
+        if (status.contains(ParticipantStatus.unconfirmed)) {
           regState = RegistrationState.PARTIALCONFIRMED;
+        } else {
+          regState = RegistrationState.CONFIRMED;
```

Comment:
why is this logic inverted? does it solve anything?

Reply:
Because a Registration with two (or more) Participants with the state "confirmed and e.g. duplicate" would return PARTIALCONFIRMED without the inversion